addon/services/permission-matrix.js

```diff
@@ -4,13 +4,19 @@ import { deprecate } from '@ember/debug';
 
 export default Service.extend({
   store: service(),
+  _permissionMatrixPromise: null,
   permissionMatrix: computed(async function(){
     deprecate('Async Computed Called', false, {id: 'common.async-computed', until: '40'});
     return this.getPermissionMatrix();
   }),
   async getPermissionMatrix() {
-    const store = this.get('store');
-    const schools = await store.findAll('school');
+    if (!this._permissionMatrixPromise) {
+      this._permissionMatrixPromise = this._fillMatrix();
```

Comment:
shouldn't the `await` be on this line?

Reply:
No because it's storing the promise itself and not the result. This keeps it non-blocking so the next call doesn't also start a request, it just moves on and waits for the one that is in flight already.

Reply:
thanks for clarifying.